packages/generator/src/templateHelpers.ts

```diff
@@ -97,6 +97,21 @@ export const isTypeDefined = function(range: any): boolean {
   return false;
 };
 
+/**
+ * Check if the type of the property is linked
+ *
+ * @param range AMF model that hold the type of the property
+ * @returns true if the type of the property is linked
+ */
+export const isTypeLinked = function(range: model.domain.Shape): boolean {
+  return (
+    range != null &&
+    range.isLink === true &&
+    range.linkTarget != null &&
+    (range.linkTarget as model.domain.ScalarShape).dataType != null
```

Comment:
Why only ScalarShape? How do we handle NodeShape which are linked?

Reply:
FotrNodeShape, we directly retrieve the type/name of the object 

export type Customer = {
    birthday?: any;
    companyName?: string;
    creationDate?: any;
    credentials?: Credentials;
    primaryAddress?: CustomerAddress;
........
} & { [key: string]: any }